libs/hacl/CMakeLists.txt

```diff
@@ -38,7 +38,12 @@ endif()
 # ---- Library Sources ----
 include(${PROJECT_SOURCE_DIR}/libs/hacl/sources.cmake)
 
-add_library(${HACL_CPP_TARGET_NAME} SHARED ${SOURCES_hacl_cpp})
+if(ELECTIONGUARD_BUILD_SHARED)
+    add_library(${HACL_CPP_TARGET_NAME} SHARED ${SOURCES_hacl_cpp})
+else()
+    add_library(${HACL_CPP_TARGET_NAME} STATIC ${SOURCES_hacl_cpp})
```

Comment:
wasm needs us to build libraries statically. it can link dynamically but it is slower and i was having issues with it